src/Traces.Core/Extensions/Files/CreateTraceFileDtoExtensions.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Collections.Generic;
+using Traces.Common;
+using Traces.Common.Constants;
+using Traces.Common.Utils;
+using Traces.Core.Models.Files;
+using Traces.Data.Entities;
+
+namespace Traces.Core.Extensions.Files
+{
+    public static class CreateTraceFileDtoExtensions
+    {
+        public static bool IsValid(this CreateTraceFileDto createTraceFileDto)
+        {
+            if (string.IsNullOrWhiteSpace(createTraceFileDto.Name) ||
+                string.IsNullOrWhiteSpace(createTraceFileDto.MimeType) ||
+                createTraceFileDto.Size <= 0 ||
+                createTraceFileDto.Size > AppConstants.MaxFileSizeInBytes ||
+                createTraceFileDto.Data == null
+            )
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public static bool IsValid(this List<CreateTraceFileDto> dtos)
+        {
+            foreach (var createTraceFileDto in dtos)
+            {
+                if (!createTraceFileDto.IsValid())
+                {
+                    return false;
+                }
+            }
+
+            return true;
```

Comment:
```suggestion
            return dtos.All(f => f.IsValid());
```

I think that might work. Do not take my word for it, it is an assumption, so maybe you can test it.